AmplifyPlugins/Auth/AWSCognitoAuthPluginIntegrationTests/AuthDeleteUserTests/AuthDeleteUserTests.swift

```diff
@@ -39,34 +40,33 @@ class AuthDeleteUserTests: AWSAuthBaseTest {
         AuthSignInHelper.registerAndSignInUser(username: username,
                                                password: password,
                                                email: email) { didSucceed, error in
-            registerAndSignInExpectation.fulfill()
+            if didSucceed {
+                registerAndSignInExpectation.fulfill()
+            }
             XCTAssertTrue(didSucceed, "SignIn operation failed - \(String(describing: error))")
```

Comment:
An alternative to checking and then asserting for success would be to defer the expectation fulfillment.  The end result would be the same in this case, but it would only require one check.  The deferral pattern would have added benefit in cases where more assertions are added because it would ensure that the expectation is only fulfilled after all assertions have been met.

Reply:
Yeah, there are some cases where we do follow this pattern (some of which I left intact). However, the failed assertions didn't cause the unit test to halt (unless `continueAfterFailure` is set to false) and those things would proceed to cause chaos after the `wait(for:timeout:)`. I may be misunderstanding - you meant something like the example below, right?

```
func testSomething() throws {
    let e = expectation(description: "foo")
    performSomethingAsync { (didSucceed, error) in
        defer {
            e.fulfill()
        }
        XCTAssertTrue(didSucceed)
    }
    wait(for: [e], timeout: 1.0)
    // If the operation above succeeded, perform the following:
   ...
}
```

Another way we could do this is save a value representing the success (e.g. `didSucceed` or `session`), then assert after waiting for the expectation (and setting `continueAfterFailure` to `false`). What do you think?

Reply:
Right, the above requires `continueAfterFailure` to be set to false if you want to end the test after a failure.  I don't have a strong opinion about which pattern to use, just pointing out alternatives.

Reply:
for the purposes of this PR, I think it's good as written